ingest-app/src/cmr/ingest/services/jobs.clj

```diff
@@ -173,10 +192,51 @@
   [context]
   (bulk-update/cleanup-old-bulk-update-status context))
 
+(defn- process-subscriptions
+  "Process each subscription in subscriptions."
+  [context subscriptions time-constraint]
+  (for [subscription subscriptions
+       :let [email-address (:email-address subscription)
+             coll-id (:collection-concept-id subscription)
+             query-params (create-query-params (:metadata subscription))
+             params1 (merge {"created-at" time-constraint}
+                            {"collection-concept-id" coll-id}
+                            query-params)
+             params2 (merge {"revision-date" time-constraint}
+                            {"collection-concept-id" coll-id}
+                            query-params)
+             gran-ref1 (search/find-granule-references context params1)
+             gran-ref2 (search/find-granule-references context params2)
+             gran-ref (distinct (concat gran-ref1 gran-ref2))]]
+      (if (seq gran-ref)
+        ;; These strings are just for debugging purpose. In another ticket
+        ;; I will figure out how to send email.
+        (str "Granules found for: " email-address)
+        (str "No granules found for: " email-address))))
+
+(defn- email-subscription-processing
+  "Process email subscriptions and send email when found granules matching the collection and queries
+  in the subscription and were created/updated during the last processing interval."
+  [context]
+  (let [end-time (t/now)
+        start-time (t/minus end-time (t/seconds (email-subscription-processing-interval)))
+        time-constraint (str (str start-time) "," (str end-time))
+        subscriptions
+         (->> (mdb/find-concepts context {:latest true} :subscription)
+              (filter #(not (:deleted %)))
+              (map #(select-keys % [:email-address :metadata :collection-concept-id])))]
+    ;; for some reason, the info or println doesn't show inside the for loop in my local
+    ;; proto repl.
+    (println (process-subscriptions context subscriptions time-constraint))))
```

Comment:
Shouldn't we use the  logging framework for this?

Reply:
yes. it's only temporary there. I had trouble getting "info" to printout inside the for loop for some reason.